crates/sui-framework/packages/sui-system/sources/sui_system.move

```diff
@@ -166,6 +166,7 @@ module sui_system::sui_system {
         sui_system_state_inner::request_remove_validator_candidate(self, ctx)
     }
 
+    #[allow(unused_mut_ref)]
```

Comment:
Copying my comment from the Slack thread here:

> Note that this lint about `&mut` to `&` currently interacts strangely with package upgrades and we may want to rethink or reword it: Someone may legitimately leave a public function with a `&mut` parameter even though it is not mutated through that reference in the current version of a package, because they know that in a future version that they will upgrade to, it will -- I think this is particularly an issue with `TxContext`.

There's also an extra dimension to this that for people who run this lint on a package they've already published and are looking to upgrade, it will advise them to perform the wrong action (break compatibility).